### PR TITLE
[WIP] Loyal slimecross extracts

### PR DESCRIPTION
**Pull request by Dingo-Dongler**
## About The Pull Request

Adds loyal slime extracts. They can be used on things to give them new properties. This is done by adding a loyal_effect component to the item. Each item can only have one loyal_effect component.

## Why It's Good For The Game

More options for xenobio. Pretty different to stuff already in xenobio.

## Changelog
:cl:
add: Loyal slime extracts can be made from light pink slime extracts. They can be used on items to give them new properties.
/:cl:

### CODE REVIEW

File: code/modules/research/xenobiology/crossbreeding/loyal.dm [hunk 1]
```diff
@@ -0,0 +1,745 @@
...
+	colour = "yellow"
+	effect_desc = "Makes an item glow."
+	loyaleffect = /datum/component/loyal_effect/yellow
+
+/datum/component/loyal_effect/yellow
+	prefix = "glowing"
+
+/datum/component/loyal_effect/yellow/Initialize(obj/item/attached_item)
+	. = ..()
+	attached_item.set_light(l_range = 5, l_power = 1) //A bit wider than a flashlight
```
Comment (Dingo-Dongler) [inline code/modules/research/xenobiology/crossbreeding/loyal.dm]: This acts funkily with items  that already emit light, like flashlights. How would I change this?

File: code/modules/research/xenobiology/crossbreeding/loyal.dm [hunk 2]
```diff
@@ -0,0 +1,745 @@
...
+	. = ..()
+	C.adjust_bodytemperature(150 - C.bodytemperature) //Stolen from cryo projectile code
+
+/datum/component/loyal_effect/darkblue/proc/equippedChanged(datum/source, mob/living/carbon/user, slot)
+	if(istype(user))
+		attached_user = user
+	else
+		attached_user = null
+
+/obj/item/slimecross/loyal/silver
```
Comment (Dingo-Dongler) [inline code/modules/research/xenobiology/crossbreeding/loyal.dm:268]: Wondering how to make this one work. Maybe something similar to moth clothe eating code?

File: code/modules/research/xenobiology/crossbreeding/loyal.dm [hunk 3]
```diff
@@ -0,0 +1,745 @@
...
+								 /obj/item/melee/baseball_bat,
+								 /obj/item/electropack,
+								 /obj/item/chair
+								)
+		if(WEIGHT_CLASS_GIGANTIC)
+			disguise_list = list(/obj/item/mecha_parts/part/honker_head, //There's barely anything of this size
+								 /obj/item/his_grace //Disguising as this is more of a laugh than a strategy
+								)
+	var/obj/item/disguise_type = pick(disguise_list)
+	var/obj/item/disguise = new disguise_type
```
Comment (Dingo-Dongler) [inline code/modules/research/xenobiology/crossbreeding/loyal.dm:632]: Any way to get the properties of disguise_type without making an instance of it?

File: code/modules/research/xenobiology/crossbreeding/loyal.dm [hunk 4]
```diff
@@ -0,0 +1,745 @@
...
+	if(ishuman(attached.loc))
+		humanfound = attached.loc
+	if(ishuman(attached.loc.loc)) //Check if in backpack.
+		humanfound = (attached.loc.loc)
+	if(!humanfound)
+		return
+	var/mob/living/carbon/human/H = humanfound
+	SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "soothing", /datum/mood_event/soothing, attached)
+
+/obj/item/slimecross/loyal/gold
```
Comment (Dingo-Dongler) [inline code/modules/research/xenobiology/crossbreeding/loyal.dm:694]: Also need to figure this one out. OnClick code? Something like tk?

File: code/modules/research/xenobiology/crossbreeding/loyal.dm [hunk 5]
```diff
@@ -0,0 +1,745 @@
...
+	blackslime.Friends[user] = 10 //This should be obediant enough
+	target.forceMove(blackslime)
+	//target.RegisterSignal(blackslime, COMSIG_MOB_DEATH, .proc/drop_attached)
+
+//mob/living/simple_animal/slime/proc/drop_attached(datum/source, gibbed)
+
+/datum/component/loyal_effect/black
+	prefix = "transformative"
+
+/obj/item/slimecross/loyal/lightpink
```
Comment (Dingo-Dongler) [inline code/modules/research/xenobiology/crossbreeding/loyal.dm:753]: I need a way to test this one, I couldn't connect with two clients to my test server.

File: code/modules/research/xenobiology/crossbreeding/loyal.dm (hunk 3 above)
Comment (Dingo-Dongler) [reply code/modules/research/xenobiology/crossbreeding/loyal.dm:632]: Also, Disguise should only be usable in hand. Same with compress in bluespace I guess. How do I check this?

File: code/modules/research/xenobiology/crossbreeding/loyal.dm [hunk 6]
```diff
@@ -0,0 +1,745 @@
...
+	. = ..()
+	if(istype(attached_item, /obj/item/clothing)) //We only care if it's clothes.
+		START_PROCESSING(SSobj,src)
+		RegisterSignal(attached_item, list(COMSIG_ITEM_EQUIPPED, COMSIG_ITEM_DROPPED), .proc/equippedChanged)
+
+/datum/component/loyal_effect/purple/Destroy()
+	STOP_PROCESSING(SSobj,src)
+	return ..()
+
+/datum/component/loyal_effect/purple/proc/equippedChanged(datum/source, mob/living/carbon/user, slot)
```
Comment (Dingo-Dongler) [inline code/modules/research/xenobiology/crossbreeding/loyal.dm:99]: If loyal purple is used on clothes while worn, it doesn't start processing until they are taken off and put back on. I gotta change that.

**Review by Dingo-Dongler — COMMENTED** [review]
Reviewing some stuff I'm struggling on, any advice is appreciated.

**Comment by PowerfulBacon** [comment]
Best bet with these is probably adding components to the item it is used on
Wait that's what you did anyway

**Comment by zeskorion** [comment]
is this done?

**Comment by Dingo-Dongler** [comment]
> 
> 
> is this done?

No, progress has been slower now that I'm back at uni. If anyone wants to finish this up I'm happy to split the bounty like 50/50.

**Pull request closed without merging**